data_pool_* symbol visibility

The new `data_pool_*` functions seem to be internal-only (or are at least undocumented?) but they appear on the library's exported symbols:
```
$ objdump -T libmaxminddb.so.0.0.7 | grep data_pool
0000000000003ce0 g    DF .text	000000000000007a  Base        data_pool_new
0000000000003e20 g    DF .text	0000000000000023  Base        data_pool_lookup
0000000000003cb0 g    DF .text	0000000000000022  Base        data_pool_destroy
0000000000003e50 g    DF .text	0000000000000054  Base        data_pool_to_list
0000000000003d60 g    DF .text	00000000000000b6  Base        data_pool_alloc
```

These can be hidden using gcc/ld tricks, like the visibility attribute or a version script with e.g. a MMDB_* pattern.

I've tried this:
```diff
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -4,7 +4,7 @@ lib_LTLIBRARIES = libmaxminddb.la
 
 libmaxminddb_la_SOURCES = maxminddb.c maxminddb-compat-util.h \
        data-pool.c data-pool.h
-libmaxminddb_la_LDFLAGS = -version-info 0:7:0
+libmaxminddb_la_LDFLAGS = -version-info 0:7:0 -export-symbols-regex '^MMDB_.*'
 include_HEADERS = $(top_srcdir)/include/maxminddb.h
 
 pkgconfig_DATA = libmaxminddb.pc
```

…but then apparently the data pool test fails with:
```
data_pool_t-data-pool-t.o: In function `test_data_pool_new':
./t/data-pool-t.c:24: undefined reference to `data_pool_new'
./t/data-pool-t.c:29: undefined reference to `data_pool_new'
./t/data-pool-t.c:34: undefined reference to `data_pool_new'
./t/data-pool-t.c:39: undefined reference to `data_pool_destroy'
[etc.]
```

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -4,7 +4,7 @@ lib_LTLIBRARIES = libmaxminddb.la
 
 libmaxminddb_la_SOURCES = maxminddb.c maxminddb-compat-util.h \
 	data-pool.c data-pool.h
-libmaxminddb_la_LDFLAGS = -version-info 0:7:0
+libmaxminddb_la_LDFLAGS = -version-info 0:7:0 -export-symbols-regex '^MMDB_.*'
 include_HEADERS = $(top_srcdir)/include/maxminddb.h
 
 pkgconfig_DATA = libmaxminddb.pc

--- a/t/Makefile.am
+++ b/t/Makefile.am
@@ -18,6 +18,7 @@ check_PROGRAMS = \
 
 data_pool_t_CFLAGS = $(CFLAGS) -I$(top_srcdir)/src
 data_pool_t_LDFLAGS = $(AM_LDFLAGS) -lm
+data_pool_t_SOURCES = data-pool-t.c ../src/data-pool.c
 
 threads_t_CFLAGS = $(CFLAGS) -pthread
 

